src/MongoDB.Driver.Core/Core/Clusters/Cluster.cs

```diff
@@ -222,9 +222,9 @@ private void RapidHeartbeatTimerCallback(object args)
 
         protected abstract void RequestHeartbeat();
 
-        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription)
+        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription, bool shouldClusterDescriptionChangedEventBePublished)
         {
-            if (_descriptionChangedEventHandler != null)
+            if (shouldClusterDescriptionChangedEventBePublished && _descriptionChangedEventHandler != null)
             {
                 _descriptionChangedEventHandler(new ClusterDescriptionChangedEvent(oldDescription, newDescription));
```

Comment:
Similarly, should we rename this to `_sdamMonitoringDescriptionChangdEventHandler`, to make it absolutely clear that this is not the sdam flow event, but the sdam monitoring flow event?